### PR TITLE
Feature: add a telemetry switch

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Create two switches for analytics and crash reports to replace the original switch for two purposes. The analytics option toggle is only shown when the self user is a team member.

### CODE REVIEW

File: Wire-iOS Tests/Analytics/AnalyticsTests.swift [hunk 1]
```diff
@@ -38,24 +38,34 @@ final class AnalyticsTests: XCTestCase {
...
     
+    func testThatItSetsOptOutAnalyticsToSharedSettings() {
+        // GIVEN
+        TrackingManager.shared.disableAnalyticsSharing = false
+        // THEN
+        XCTAssertFalse(ExtensionSettings.shared.disableAnalyticsSharing)
+        // WHEN
+        TrackingManager.shared.disableAnalyticsSharing = true
+        // THEN
+        XCTAssert(ExtensionSettings.shared.disableAnalyticsSharing)
```
Comment (johnxnguyen) [inline Wire-iOS Tests/Analytics/AnalyticsTests.swift:69]: `XCTAssertTrue`?

File: WireCommonComponents/ExtensionSettings.swift [hunk 2]
```diff
@@ -19,28 +19,27 @@
...
+    case disableCrashSharing = "disableCrashSharing"
+    case disableAnalyticsSharing = "disableAnalyticsSharing"
 
-    static var all: [ExtensionSettingsKey] {
-        return [
-            .disableLinkPreviews,
-            .disableCrashAndAnalyticsSharing
-        ]
-    }
+    case disableLinkPreviews = "disableLinkPreviews"
```
Comment (johnxnguyen) [inline WireCommonComponents/ExtensionSettings.swift:27]: The explicit raw values are unnecessary.

File: Wire-iOS Tests/Analytics/AnalyticsTests.swift (hunk 1 above)
Comment (billypchan) [reply Wire-iOS Tests/Analytics/AnalyticsTests.swift:69]: They are the same.

File: WireCommonComponents/ExtensionSettings.swift (hunk 2 above)
Comment (billypchan) [reply WireCommonComponents/ExtensionSettings.swift]: cleaned up old redundant code

File: Wire-iOS Tests/Analytics/AnalyticsTests.swift (hunk 1 above)
Comment (johnxnguyen) [reply Wire-iOS Tests/Analytics/AnalyticsTests.swift:69]: I know, but I guess it's worth being consistent with the other assertions.

Comment (billypchan) [reply Wire-iOS Tests/Analytics/AnalyticsTests.swift:69]: cleaned up other `XCTAssertTrue` in this file.

**Review by johnxnguyen — APPROVED** [review]

File: Wire-iOS Tests/Analytics/AnalyticsTests.swift (hunk 1 above)
Comment (marcomaddy1982) [reply Wire-iOS Tests/Analytics/AnalyticsTests.swift:65]: I guess he means that they should be consistent with the other assertions in the entire test files.

**Pull request merged**